project/Modules.scala

```diff
@@ -0,0 +1,10 @@
+import sbt.*
+
+object Modules {
+  val CirceVersion = "0.14.6"
+
+  lazy val circe: List[ModuleID] =
+    "io.circe"   %% "circe-parser"  % CirceVersion ::
```

Comment:
Why is this written this way? It's more idiomatic IME to use the `List` constructor.

Reply:
By this way, I assume you are referring to the use of `::` (and `:::`).

Let me explain. Not sure if it is branded idiomatic but AFAIK `libraryDependencies` are declared using `Seq`. I am more a fan of using the `::` syntax. 

... because the following is a bit noisy to me .... especially when we start more dependencies.

```scala
val dependencies = Seq(
  "io.circe"   %% "circe-parser"  % CirceVersion,
  "io.circe" %% "circe-generic" % CirceVersion,
  ...
)
```

The following is less so and nice with scalafmt arranging the parts of the module def nicely in columns (easier for eyes to navigate):

```scala
val dependencies =
  "io.circe"   %% "circe-parser"  % CirceVersion ::
    "io.circe" %% "circe-generic" % CirceVersion ::
    Nil
```

Add to the above, on a higher scale, it makes project files easier to maintain when you _group_ and declare dependencies like how I have done with `Circe`, and then aggregate them.

We would do this for test dependencies:

```scala
val testDependencies =
  "org.scalatest" %% "scalatest" % "3.2.15" % Test ::
    "org.scalacheck" %% "scalacheck" % "1.17.0" ::
    .... ::
    Nil
```


Typically, you would do this for compile time dependencies

```scala
val krop =
  ... all krop related dependencies ... ::
    Nil

// I know we get this transitively. I can't think of other dependencies as I am writing this.
val cats =
  "org.typelevel" %% "cats-core" % "2.9.0" ::
    "org.typelevel" %% "cats-free" % "2.9.0" ::
    ...
    Nil
  

val devDependencies =
  krop ::: cats ::: ....
   
```

Finally in build.sbt:

```scala
lazy val snorriRoot =
  project
    .in(file("."))
    .settings(
      ....,
      libraryDependencies ++=
         // You could use devDependencies or break down the dependencies as declared in Modules. I typically prefer devDependencies.
         Modules.devDependencies ++ 
           Modules.testDependencies
    )

```


This has worked quite well for me for some large sized projects I have worked on - writing and maintaining build files (especially when there are changes, it is easy to see what actually changed in the build definition).

That is my lonnnngg answer 😃 . _I am happy to adjust the style based on your feedback_.

Reply:
I see. Thanks for the explanation. That makes sense. I'd probably be inclined to use the `Seq` methods `+:` and `++`, as I think they are more widely known, but this is very minor.

Reply:
`+:` is nice but ...
- It is generally always confusing for a lot of people if it is `+:` or `:+` (even though the `+` determines which one to use).
- Also, I am not sure if you can use `+:` or `:+` without first starting with a `Seq` unlike `List` where you just need `::` and end with a `Nil`.

Let me know if you are okay to keep `::` and see how it works out for us. If not, I will switch it to `+:` and `Seq`.